### ISSUE TITLE
What are the use cases for this?

### ISSUE THREAD

**Issue by mudrd8mz** [hitteshahuja/filter_bath_https_monitor#3]
Can you please elaborate a bit more on the actual use cases that led to developing this? In many (if not most) cases, users themselves do not embed content via iframe/embed tags, but such inclusion is a result of some filter. It is not clear from the user's perspective what should I do to "make sure embedded content on this site is HTTPS".

Additionally, such a warning seems to be displayed to all users, even if they were not authors of the text?

**Comment by hitteshahuja** [comment]
@mudrd8mz Good Question! Here is why I thought there would be a need for the plugin : 

We recently moved from in-house hosting to external hosting whereby our Moodle is hosted by AWS and managed by an external company. We also decided to go https for the whole site this time rather than just https login. Little did we realise that every non-https content on the site would need to https or else they would give a warning on some browsers( like IE ) and on some browsers they would display blank but with an option to "unlock" insecure content. This plugin ( I hope) aims to notify users to direct to only secure content wherever possible. You are right in pointing out that this msg should only be displayed to users who have the right to edit content like teachers, admins etc. Students would not care or benefit from that message. We considered using the find and replace tool(https://docs.moodle.org/22/en/Search_and_replace_tool)` to convert basic urls from http to https but need a sustainable solution that would atleast try and enforce best practice among teachers to upload only secure content. We also considered implementing the w3c security navigational upgrades (https://www.w3.org/TR/upgrade-insecure-requests/#example-navigation) but this is the last resort we would like to come to if all else fails .
